sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/data/cosmos/CosmosDataDiagnosticsConfigurationTests.java

```diff
@@ -0,0 +1,95 @@
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.*;
+
+class CosmosDataDiagnosticsConfigurationTests {
+
+    private static final String ENDPOINT = "https://test.documents.azure.com:443/";
+    private final ApplicationContextRunner contextRunner = new ApplicationContextRunner()
+        .withConfiguration(AutoConfigurations.of(
+            AzureGlobalPropertiesAutoConfiguration.class,
+            AzureCosmosAutoConfiguration.class,
+            CosmosDataAutoConfiguration.class));
+    CosmosClientBuilder cosmosClientBuilder = mock(CosmosClientBuilder.class);
+    CosmosAsyncClient cosmosAsyncClient = mock(CosmosAsyncClient.class);
+
+    @Test
+    void configureWithPopulateQueryMetricsEnabled() {
+        try (MockedStatic<CosmosFactory> mockedStatic = mockStatic(CosmosFactory.class, RETURNS_MOCKS)) {
+            when(cosmosClientBuilder.buildAsyncClient()).thenReturn(cosmosAsyncClient);
+            mockedStatic.when(() -> CosmosFactory.createCosmosAsyncClient(cosmosClientBuilder))
+                .thenReturn(cosmosAsyncClient);
+            this.contextRunner
+                .withBean(CosmosClientBuilder.class, () -> cosmosClientBuilder)
+                .withBean(CosmosClient.class, () -> mock(CosmosClient.class))
+                .withPropertyValues(
+                    "spring.cloud.azure.cosmos.endpoint=" + ENDPOINT,
+                    "spring.cloud.azure.cosmos.database=test",
+                    "spring.cloud.azure.cosmos.populate-query-metrics=true")
+                .run(context -> assertThat(context).hasSingleBean(ResponseDiagnosticsProcessor.class));
+        }
+    }
+
+    @Test
+    void configureWithPopulateQueryMetricsDisabled() {
+        try (MockedStatic<CosmosFactory> mockedStatic = mockStatic(CosmosFactory.class, RETURNS_MOCKS)) {
+            when(cosmosClientBuilder.buildAsyncClient()).thenReturn(cosmosAsyncClient);
+            mockedStatic.when(() -> CosmosFactory.createCosmosAsyncClient(cosmosClientBuilder))
+                .thenReturn(cosmosAsyncClient);
+            this.contextRunner
+                .withBean(CosmosClientBuilder.class, () -> cosmosClientBuilder)
+                .withBean(CosmosClient.class, () -> mock(CosmosClient.class))
+                .withPropertyValues(
+                    "spring.cloud.azure.cosmos.endpoint=" + ENDPOINT,
+                    "spring.cloud.azure.cosmos.database=test-database",
+                    "spring.cloud.azure.cosmos.populate-query-metrics=false")
+                .run(context -> assertThat(context).doesNotHaveBean(ResponseDiagnosticsProcessor.class));
+        }
+    }
```

Comment:
Can we have an assertion that without the ResponseDiagnosticsProcessor bean, spring data cosmos can be configured correctly?